ocs_archive/input/file.py

```diff
@@ -97,6 +102,36 @@ def __init__(self, open_file: File, file_metadata: dict = None, blacklist_header
         self._repair_observation_day()
         self._repair_public_date()
 
+    @property
+    @lru_cache()
```

Comment:
Seemed smart to cache this. @lru_cache exists throughout the lifetime of the instance - which should be short enough that we shouldn't have to do any cache invalidation? there is the case where tags are updated, but I think that'll be rare enough that it likely won't be an issue, and end-users can override this if needed.